### PR TITLE
Batch norm cudnn accurate

### PR DESCRIPTION
**Pull request by typhoonzero**
### ***IMPORTANT BUG FIX***

This will fix accuracy issues when training imagenet for face classification models using large batch sizes.

**NOTE:** this fix has been tested on training imagenet using `batch_size=96*48` or larger.

See cudnn docs from: https://docs.nvidia.com/deeplearning/sdk/cudnn-developer-guide/index.html#cudnnQueryRuntimeError

> When the CUDNN_BATCHNORM_SPATIAL_PERSISTENT mode is selected in cudnnBatchNormalizationForwardTraining or cudnnBatchNormalizationBackward, the algorithm may encounter numerical overflows where CUDNN_BATCHNORM_SPATIAL performs just fine albeit at a slower speed. The user can invoke cudnnQueryRuntimeError to make sure numerical overflows did not occur during the kernel execution. Those issues are reported by the kernel that performs computations.

This is set to false by default also in other frameworks like: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/cuda/cuda_dnn.cc#L622

### CODE REVIEW

**Review by Yancey0623 — DISMISSED** [review]
LGTM+++

**Review by chengduoZH — COMMENTED** [review]
Do you have a unit test for this?

**Review by chengduoZH — DISMISSED** [review]

**Comment by typhoonzero** [comment]
@chengduoZH Will try add this in another PR, should need to somehow construct data cause the overflow.

**Comment by qingqing01** [comment]
We also found this problem.  https://github.com/PaddlePaddle/Paddle/pull/16243  @heavengate Please see this PR.

**Comment by heavengate** [comment]
我试过在  #16243 disable了这个模式，合入之后image classification的CE eval_loss超限了，现在flag是默认是false的话有可能会再次出现这个问题

**Review by Yancey0623 — APPROVED** [review]
lgtm

**Comment by Yancey0623** [comment]
Should confirm this PR on CE.
